packages/jaeger-ui/src/components/common/CopyIcon.test.js

```diff
@@ -13,58 +13,57 @@
 // limitations under the License.
 
 import React from 'react';
-import { shallow } from 'enzyme';
-import { Button, Tooltip } from 'antd';
+import { screen, render, waitFor, fireEvent, waitForElementToBeRemoved } from '@testing-library/react';
+import '@testing-library/jest-dom';
+
 import * as copy from 'copy-to-clipboard';
 
 import CopyIcon from './CopyIcon';
 
 jest.mock('copy-to-clipboard');
 
 describe('<CopyIcon />', () => {
+  const tooltipTitle = 'testTooltipTitleValue';
+
   const props = {
     className: 'classNameValue',
     copyText: 'copyTextValue',
-    tooltipTitle: 'tooltipTitleValue',
+    tooltipTitle,
   };
   let copySpy;
-  let wrapper;
 
   beforeAll(() => {
     copySpy = jest.spyOn(copy, 'default');
   });
 
   beforeEach(() => {
     copySpy.mockReset();
-    wrapper = shallow(<CopyIcon {...props} />);
+    render(<CopyIcon {...props} />);
   });
 
   it('renders as expected', () => {
-    expect(wrapper).toMatchSnapshot();
+    expect(screen.getByRole('button'));
   });
 
-  it('updates state and copies when clicked', () => {
-    expect(wrapper.state().hasCopied).toBe(false);
+  it('updates state and copies when clicked', async () => {
+    const triggerElement = screen.getByRole('button');
+
+    // Intially when we hover over the icon, there should be nothing in the clipboard and the tooltip should have the default title
+    fireEvent.mouseEnter(triggerElement);
+    await waitFor(() => expect(screen.getByText(tooltipTitle)).toBeInTheDocument());
```

Comment:
what does waitFor do - is it periodically polling its lambda? Or does expect() always return a future? It's a bit strange that you suddenly start calling function in async which normally was not called this way.

Is `fireEvent.mouseEnter` sync or async?

Reply:
`waitFor` returns a Promise that resolves when the specified condition is met. It retries the condition every `50ms` (by default) until it passes or times out after `1000ms` (by default).

`fireEvent.mouseEnter` is a sync event, but even after the occurrence of the event, `antd` takes some time to insert the markdown for the tooltip into DOM. Thus we need to make use of the `waitFor` utility.